HazelcastCloudTests/csharphazelcastcloudtests/CloudTests/Helper.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+// http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.IO;
+using System.Security.Authentication;
+using System.Threading.Tasks;
+using Hazelcast;
+using Hazelcast.DistributedObjects;
+using Hazelcast.Networking;
+using Hazelcast.Testing.Logging;
+using Microsoft.Extensions.Logging;
+using NUnit.Framework;
+
+namespace CloudTests
+{
+    public class Helper
+    {
+        public static HazelcastOptions CreateClientConfigWithSsl(string nameForConnect, string token, bool isSmartClient, string certificatesPath, string tlsPassword)
+        {
+            var options = CreateClientConfigWithoutSsl(nameForConnect, token, isSmartClient);
+            options.Networking.Ssl.Enabled = true;
+            options.Networking.Ssl.ValidateCertificateChain = false;
+            options.Networking.Ssl.Protocol = SslProtocols.Tls12;
+            options.Networking.Ssl.CertificatePath = Path.GetFullPath(certificatesPath + "client.pfx");
+            options.Networking.Ssl.CertificatePassword = tlsPassword;
+            return options;
+        }
+
+        public static HazelcastOptions CreateClientConfigWithoutSsl(string nameForConnect, string token, bool isSmartClient)
+        {
+            var options = new HazelcastOptionsBuilder().WithHConsoleLogger()
+                .With("Logging:LogLevel:Hazelcast", "WARNING")
+                .Build();
+            options.ClusterName = nameForConnect;
+            options.Networking.Cloud.DiscoveryToken = token;
+            options.Networking.ReconnectMode = ReconnectMode.ReconnectAsync;
```

Comment:
Why did we set this to async?

Reply:
There is no difference in reconnect modes in .Net. So, sync and async behaves same.

Reply:
There should be quite a bit of distinction between these two modes but that is a topic for another discussion 😄